src/protocols/strategies/balancer.strategy.ts

```diff
@@ -215,18 +212,25 @@ export async function getBalancerSwapFees(vault: VaultDefinitionModel): Promise<
       return [];
     }
 
+    const measuredLength = poolSnapshots.length - 1;
+
     let totalFees = 0;
-    let totalLiquidity = 0;
+    let averageLiquidity = 0;
 
-    for (const snapshot of poolSnapshots) {
-      const { swapFees, liquidity } = snapshot;
-      totalFees += Number(swapFees);
-      totalLiquidity += Number(liquidity);
+    for (let i = 0; i < measuredLength; i++) {
+      const { swapFees, liquidity } = poolSnapshots[i];
+      const { swapFees: previousSwapFees } = poolSnapshots[i + 1];
+      totalFees += Number(swapFees) - Number(previousSwapFees);
+      averageLiquidity += Number(liquidity);
     }
 
+    averageLiquidity /= measuredLength;
+    // balancer takes 50% of fees
+    totalFees /= 2;
```

Comment:
can this and averageLiquidity be changed to new variables like "balancerFee = totalFee / 2", its a preference but i try to avoid mutations